src/java.base/share/classes/java/net/SocketOptions.java

```diff
@@ -47,224 +47,169 @@
      */
-
     @Native public static final int SO_BROADCAST = 0x0020;
 
-    /** Set which outgoing interface on which to send multicast packets.
-     * Useful on hosts with multiple network interfaces, where applications
-     * want to use other than the system default.  Takes/returns an InetAddress.
-     * <P>
-     * Valid for Multicast: DatagramSocketImpl
+    /**
+     * See {@link StandardSocketOptions#IP_MULTICAST_IF} for description of this socket option.
      *
      * @see MulticastSocket#setInterface(InetAddress)
      * @see MulticastSocket#getInterface()
      */
-
     @Native public static final int IP_MULTICAST_IF = 0x10;
 
-    /** Same as above. This option is introduced so that the behaviour
-     *  with IP_MULTICAST_IF will be kept the same as before, while
-     *  this new option can support setting outgoing interfaces with either
-     *  IPv4 and IPv6 addresses.
+    /**
+     * This option is used to both set and fetch the outgoing interface on which the multicast
+     * packets are sent. Useful on hosts with multiple network interfaces, where applications
+     * want to use other than the system default. This option supports setting outgoing interfaces
+     * with either IPv4 and IPv6 addresses.
      *
-     *  NOTE: make sure there is no conflict with this
      * @see MulticastSocket#setNetworkInterface(NetworkInterface)
      * @see MulticastSocket#getNetworkInterface()
      * @since 1.4
      */
     @Native public static final int IP_MULTICAST_IF2 = 0x1f;
 
     /**
-     * This option enables or disables local loopback of multicast datagrams.
-     * This option is enabled by default for Multicast Sockets.
+     * See {@link StandardSocketOptions#IP_MULTICAST_LOOP} for description of this socket option.
      * @since 1.4
      */
-
     @Native public static final int IP_MULTICAST_LOOP = 0x12;
 
     /**
-     * This option sets the type-of-service or traffic class field
-     * in the IP header for a TCP or UDP socket.
+     * See {@link StandardSocketOptions#IP_TOS} for description of this socket option.
      * @since 1.4
      */
-
     @Native public static final int IP_TOS = 0x3;
 
     /**
-     * Specify a linger-on-close timeout.  This option disables/enables
-     * immediate return from a <B>close()</B> of a TCP Socket.  Enabling
-     * this option with a non-zero Integer <I>timeout</I> means that a
-     * <B>close()</B> will block pending the transmission and acknowledgement
-     * of all data written to the peer, at which point the socket is closed
-     * <I>gracefully</I>.  Upon reaching the linger timeout, the socket is
-     * closed <I>forcefully</I>, with a TCP RST. Enabling the option with a
-     * timeout of zero does a forceful close immediately. If the specified
-     * timeout value exceeds 65,535 it will be reduced to 65,535.
-     * <P>
-     * Valid only for TCP: SocketImpl
+     * See {@link StandardSocketOptions#SO_LINGER} for description of this socket option.
```

Comment:
One thing to mention about SO_LINGER is that the value of this socket option is an Integer when enabling and a Boolean when disabling. The getter works the same way. AFAIK, this was never correctly specified in SocketOptions.SO_LINGER but the examples in the setOption/getOption API docs include disabling. To notice this would require something to use a custom SocketImpl of course. So I think we need a follow-up to the change here to say that the value is Boolean.FALSE to disable (or disabled), otherwise an Integer with the linger timeout.